Add support for the Feature unsupported error

RonEld opened the pull request:
Add support for `MBEDTLS_ERR_PLATFORM_FEATURE_UNSUPPORTED`.
If the underlying platform does not support the current feature,
skip the test, instead of failing.

Dependent on https://github.com/ARMmbed/mbed-os/pull/8643 and https://github.com/ARMmbed/mbedtls/pull/2054 to be merged

RonEld commented:
Both the preceding PRs have been merged
cc @k-stachowiak

k-stachowiak commented on benchmark/main.cpp:
```diff
@@ -1051,22 +1085,29 @@ MBED_NOINLINE static int benchmark_ecdsa()
-
         ret = mbedtls_snprintf(title, sizeof(title), "ECDSA-%s",
-                               curve_info->name);
+                                       curve_info->name);
```
It seems that this line is not indented correctly.

k-stachowiak commented on benchmark/main.cpp:
```diff
@@ -1135,33 +1190,56 @@ MBED_NOINLINE static int benchmark_ecdh()
-                           "mbedtls_snprintf(): %d\n", ret);
-            goto exit;
-        }
+
```
This empty line is redundant.

k-stachowiak commented on benchmark/main.cpp:
```diff
@@ -1246,15 +1353,36 @@ MBED_NOINLINE static int benchmark_ecdh_curve22519()
     mbedtls_ecdh_init(&ecdh);
     mbedtls_mpi_init(&z);
 
+    ret = mbedtls_snprintf(title, sizeof(title), "ECDHE-Curve25519");
```
This line is misaligned.

RonEld commented on benchmark/main.cpp:
```diff
@@ -184,7 +184,11 @@ do {                                                    \
                                                         \
-    if (ret != 0) {                                     \
+    if (ret == MBEDTLS_ERR_PLATFORM_FEATURE_UNSUPPORTED)\
+    {                                                   \
```
I'm sorry, but I don't understand. What brace? What condition?

k-stachowiak replied:
As far as I know, in Mbed OS (or at least in our example applications) the braces should be on the same lines as the if condition. So it should be:

    if (condition) {

rather than:

    if (condition)
    {

My mind may have been clouded when I wrote the comment, because I agree it isn't clear. Most probably I wanted to write "... on the same line as the condition _above_".

RonEld commented:
@k-stachowiak Thank you for the review!

I have addressed your comments.

As for your note:
>note that I do not know, in which functions the "feature anavailable" error is acceptable, and in which hot; therefore I did not evaluate this aspect

The underlying platform can return this error on *any* function that it implements. For uniformity, I added a check for this error for all crypto errors, even if they don't have an alternative implementation for the time being.
For reference, you can look at https://github.com/ARMmbed/mbedtls/pull/2124 for the changes in the benchmark application.

k-stachowiak commented:
@RonEld Thanks for the changes that you have provided. I'm sorry I wasn't clear in my previous review; I am afraid that some more style changes should be made in this PR. What I mean is having the opening and closing braces on the same lines as the statements. To avoid confusion I linked the Mbed OS coding style paragraph in the review comment.

RonEld commented:
@k-stachowiak Thank you for refering me to hte mbed OS coding style.
I ha ve modified the craces style in the file.
please review
